src/Sil/Component/Product/Model/ProductVariant.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * Copyright (C) 2015-2018 Libre Informatique
+ *
+ * This file is licenced under the GNU LGPL v3.
+ * For the full copyright and license information, please view the LICENSE.md
+ * file that was distributed with this source code.
+ */
+
+namespace Sil\Component\Product\Model;
+
+use Blast\Component\Resource\Model\ResourceInterface;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\Common\Collections\ArrayCollection;
+
+class ProductVariant implements ResourceInterface, ProductVariantInterface
+{
+    /**
+     * Product unique identifyer code.
+     *
+     * @var string
+     */
+    protected $code;
+
+    /**
+     * Product name.
+     *
+     * @var string
+     */
+    protected $name;
+
+    /**
+     * Product is enabled (available).
+     *
+     * @var bool
+     */
+    protected $enabled = true;
+
+    /**
+     * The Product associated with this variant.
+     *
+     * @var Product
+     */
+    protected $product;
+
+    /**
+     * Collection of Options.
+     *
+     * @var Collection|Option[]
+     */
+    protected $options;
+
+    public function __construct(ProductInterface $product, $code, $name)
+    {
+        $this->options = new ArrayCollection();
+
+        $this->product = $product;
+        $this->code = $code;
+        $this->name = $name;
+    }
+
+    /**
+     * @return string
+     */
+    public function getCode(): string
+    {
+        return $this->code;
+    }
+
+    /**
+     * @param string $code
+     */
+    public function setCode(string $code): void
+    {
+        $this->code = $code;
+    }
+
+    /**
+     * @return string
+     */
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    /**
+     * @param string $name
+     */
+    public function setName(string $name): void
+    {
+        $this->name = $name;
+    }
+
+    /**
+     * @return bool
+     */
+    public function getEnabled(): bool
+    {
+        return $this->enabled;
+    }
+
+    /**
+     * @param bool $enabled
+     */
+    public function setEnabled(bool $enabled): void
+    {
+        $this->enabled = $enabled;
+    }
+
+    /**
+     * @return Product
+     */
+    public function getProduct(): Product
+    {
+        return $this->product;
+    }
+
+    /**
+     * @param Product $product
+     */
+    public function setProduct(Product $product): void
```

Comment:
Can we actually change the Product of a Variant ? I think not.

Reply:
No we need to disallow this kind of change.
Removing the setter